projects/angular-auth-oidc-client/src/lib/interceptor/auth.interceptor.ts

```diff
@@ -16,46 +16,68 @@ export class AuthInterceptor implements HttpInterceptor {
   ) {}
 
   intercept(req: HttpRequest<any>, next: HttpHandler): Observable<HttpEvent<any>> {
-    if (!this.configurationService.hasAtLeastOneConfig()) {
-      return next.handle(req);
-    }
-
-    const allConfigurations = this.configurationService.getAllConfigurations();
-    const allRoutesConfigured = allConfigurations.map((x) => x.secureRoutes || []);
-    const allRoutesConfiguredFlat = [].concat(...allRoutesConfigured) as string[];
-
-    if (allRoutesConfiguredFlat.length === 0) {
-      this.loggerService.logDebug(allConfigurations[0], `No routes to check configured`);
+    return interceptRequest(req, next.handle, {
+      configurationService: this.configurationService,
+      authStateService: this.authStateService,
+      closestMatchingRouteService: this.closestMatchingRouteService,
+      loggerService: this.loggerService,
+    });
+  }
+}
 
-      return next.handle(req);
-    }
+export function authInterceptor(): HttpInterceptorFn {
+  return (req, next) => {
+    return interceptRequest(req, next, {
+      configurationService: inject(ConfigurationService),
+      authStateService: inject(AuthStateService),
+      closestMatchingRouteService: inject(ClosestMatchingRouteService),
+      loggerService: inject(LoggerService),
+    });
+  };
+}
 
-    const { matchingConfig, matchingRoute } = this.closestMatchingRouteService.getConfigIdForClosestMatchingRoute(
-      req.url,
-      allConfigurations
-    );
+function interceptRequest(
```

Comment:
I don't know if there's a better way, but I extracted the logic into this `interceptRequest` method so it can be reused within the class interceptor and the functional interceptor.

Reply:
Yes, that looks fine to me. Thanks. I think I would have done it the same way.